pkg/apis/aws/validation/infrastructure.go

```diff
@@ -85,6 +86,16 @@ func ValidateInfrastructureConfig(infra *apisaws.InfrastructureConfig, nodesCIDR
 		allErrs = append(allErrs, field.Required(networksPath.Child("zones"), "must specify at least the networks for one zone"))
 	}
 
+	if len(infra.Networks.VPC.GatewayEndpoints) > 0 {
```

Comment:
Can we have a unit test for this?

Reply:
in progress .. why is the voter "concourse-ci/publish" failing on helm stuff?

Reply:
```
Executing check-generate
go generate needs to be run:
 M pkg/apis/aws/v1alpha1/zz_generated.conversion.go
 M pkg/apis/aws/zz_generated.deepcopy.go
```

Can you read the logs? They should be public.

Reply:
I've run both ‘make generate‘ and ’go generate’, still getting this failure ..

Reply:
Currently `make generate` on your PR adds to following changes:
```
$ git status
On branch gatewayendpoints
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   pkg/apis/aws/v1alpha1/zz_generated.conversion.go
	modified:   pkg/apis/aws/zz_generated.deepcopy.go

no changes added to commit (use "git add" and/or "git commit -a")
```

You need to add them as part of your PR.

Reply:
What is the `make generate` output for you?

Reply:
```text
> Generate
Successfully generated controller registration at ../../example/controller-registration.yaml
Generating deepcopy funcs
Generating defaulters
Generating conversions
E0209 13:23:52.260094    5469 conversion.go:667] Warning: could not find nor generate a final Conversion function for github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws.VPC -> github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/v1alpha1.VPC
E0209 13:23:52.262090    5469 conversion.go:668]   the following fields need manual conversion:
E0209 13:23:52.267350    5469 conversion.go:670]       - GatewayEndpoints
Generating deepcopy funcs
Generating defaulters
Generating conversions
I0209 13:27:53.975702    8837 main.go:123] parsing go packages in directory .
I0209 13:28:38.693263    8837 main.go:225] using package=.
I0209 13:28:38.920095    8837 main.go:161] written to ../../../../hack/api-reference/api.md
I0209 13:28:39.383321    9530 main.go:123] parsing go packages in directory .
I0209 13:29:35.271869    9530 main.go:225] using package=.
I0209 13:29:35.434090    9530 main.go:161] written to ../../../../hack/api-reference/config.md
```

Reply:
See the comment from Rafael above https://github.com/gardener/gardener-extension-provider-aws/pull/17/files#r374106445. 
You have to extend the versioned API for the VPC also: https://github.com/gardener/gardener-extension-provider-aws/blob/641a4095314420a302d165c24875caebcfc1b66a/pkg/apis/aws/v1alpha1/types_infrastructure.go#L86-L94, too

Reply:
@vpnachev , @ialidzhikov , i've done this already
https://github.com/gardener/gardener-extension-provider-aws/blob/5d2344a44a1103bc7c1f97ae1238e2ddfa146191/pkg/apis/aws/v1alpha1/types_infrastructure.go#L96

Reply:
@vpnachev , @ialidzhikov , done